### ISSUE TITLE
Add from_cfn option to storage.volumes.<volume>.efs.id

### ISSUE THREAD

**Issue by ghost** [aws/copilot-cli#5132]
# Problem statement

I have followed [Creating Mount Targets (CloudFormation)](https://aws.github.io/copilot-cli/docs/developing/storage/#cloudformation) guide to create an EFS volume and the corresponding mount targets. To be able to use this filesystem I created an CloudFormation export with the filesystem's ID just as the article suggested:

```yaml
Resources:
  # ...
Outputs:
  EFSVolumeID:
    Value: !Ref EFSFileSystem
    Export:
      Name: !Sub ${App}-${Env}-FilesystemID # <== EFS ID exported here
```

_Note: I use this CloudFormation script as an environment addon_

Now comes the big question: how do I use the dynamic value of this CloudFormation export in my manifest? It seems to me that the only way to define an external EFS volume in the manifest is to hardcode its value as the [`storage.volumes.<volume>.efs.id`](https://aws.github.io/copilot-cli/docs/manifest/backend-service/#id) field in the manifest:

```yaml
storage:
  volumes:
    myEFSVolume: # This is a variable key and can be set to an arbitrary string.
      path: '/etc/mount1'
      efs:
        id: fs-1234567 # <== I would like to use the value of the CF Export
```

# Proposal
Just like #4220 is a solution for a similar problem (security groups, variables and secrets), the same syntax could be adapted for the efs id:
```yaml
storage:
  volumes:
    myEFSVolume: # This is a variable key and can be set to an arbitrary string.
      path: '/etc/mount1'
      efs:
        id:
          from_cfn: ${App}-${Env}-FilesystemID # <== I would like to use the value of the CF Export
```

# Workaround
I create the EFS volume (which is an environment addon) first, note down its ID and add it to the [`environments`](https://aws.github.io/copilot-cli/docs/manifest/backend-service/#environments) map as an override to the manifest:

```yaml
environments:
  myenv:
    storage:
      volumes:
        myEFSVolume:
          efs:
            id: fs-XXXXXXXXXX
```

**Comment by efekarakus** [comment]
Thank you so much for such a well structured feature request! Your proposal makes a lot sense 👍

**Comment by dannyrandall** [comment]
Hey @bencehornak-gls! We just released this feature in [v1.30.0](https://github.com/aws/copilot-cli/releases/tag/v1.30.0)!🚀🎉

**Comment by ghost** [comment]
Awesome! Thank you for the very quick resolution!
